python/beeai_framework/workflows/agent.py

```diff
@@ -69,8 +69,24 @@ def del_agent(self, name: str) -> "AgentWorkflow":
 
     def add_agent(
         self,
-        agent: (BaseAgent | Callable[[ReadOnlyMemory], BaseAgent | asyncio.Future[BaseAgent]] | AgentFactoryInput),
+        agent: (
+            BaseAgent | Callable[[ReadOnlyMemory], BaseAgent | asyncio.Future[BaseAgent]] | AgentFactoryInput | None
+        ) = None,
+        /,
+        **kwargs: Any,
     ) -> "AgentWorkflow":
+        if agent is None or isinstance(agent, dict):
```

Comment:
You can simplify this.

1. `agent` is non-empty and kwargs is non-empty -> throw
2. `agent` is non-empty and kwargs is empty -> valid
3. `agent` is empty and kwargs is empty -> throw
4. `agent` is empty and kwargs is non-empty -> interpreter kwargs as `AgentFactoryInput`

Reply:
https://github.com/i-am-bee/beeai-framework/pull/518/commits/8c8b55e846caf0d26dd4930f9f70505defc84124